main.js

```diff
@@ -0,0 +1,286 @@
+  try {
+    mapboxgl.accessToken = 'pk.eyJ1Ijoia3JpbGlhbjg4IiwiYSI6ImNrbHc2bnR2NTBoZmYycG82dXNqaGJpZ28ifQ.XFI7bCUjMRXp6_KZvfePOg'
+    const map = new mapboxgl.Map({
+      container: 'map',
+      style: 'mapbox://styles/mapbox/streets-v11',
+      zoom: 11,
+      center: [lng, lat],
+    });
+  } catch (e) {
+    getError(searchInput, `ERROR(${e.code}): ${e.message}`);
+  }
+}
+
+function changeLanguage (event) {
+  switch (event.target.options[event.target.selectedIndex].value) {
+    case 'en':
+      language = 'en'
+      getPositionByOpenCage()
+      break
+    case 'ru':
+      language = 'ru'
+      getPositionByOpenCage()
+      break
+  }
+}
+
+function selectTemperature(event) {
+  degreesButtons.forEach((elem) => {
+    elem.classList.remove('active')
+  })
+  event.target.classList.add('active')
+  switch (event.target.dataset.temp) {
+    case 'imperial':
+      localStorage.setItem('temperature', 'imperial')
+      measures = 'imperial'
+      windSpeed = languageOptions.value === 'en' ? 'miles/h' : 'миль/ч'
+      getPositionByOpenCage()
+      break
+    case 'metric':
+      localStorage.setItem('temperature', 'metric')
+      measures = 'metric'
+      windSpeed = languageOptions.value === 'en' ? 'meters/s' : 'метров/c'
+      getPositionByOpenCage()
+      break
+  }
+}
+
+function localStorageHandler() {
+  languageOptions.value = localStorage.getItem('language') || 'en'
+  measures = localStorage.getItem('temperature') || 'imperial'
+  degreesButtons.forEach((elem) => {
+    if (elem.dataset.temp === measures) {
+      elem.classList.add('active')
+    }
+  })
+}
+
+function selectLanguagePattern(feels_like, humidity,description,wind) {
+  switch(languageOptions.value) {
+    case 'ru':
+    localStorage.setItem('language', 'ru')
+    setDateAndTime()
+    mapLatitude.innerHTML = `<span>Широта: ${mapLatitude.textContent}</span>`
+    mapLongitude.innerHTML = `<span>Долгота: ${mapLongitude.textContent}</span>`
+    windSpeed = measures === 'imperial' ? 'миль/ч' : 'м/с'
+    searchInput.placeholder = 'Поиск города'
+    searchButton.innerText = 'поиск'
+    todayDescription.innerHTML = `<span>${description}</span>
+          <span>ощущается: ${Math.round(feels_like)}°</span>
+          <span>ветер: ${wind} ${windSpeed}</span>
+          <span>влажность: ${humidity}%</span>`
+      break
+    case 'en':
+    localStorage.setItem('language', 'en')
+    setDateAndTime()
+    mapLatitude.innerHTML = `<span>Latitude: ${mapLatitude.textContent}</span>`
+    mapLongitude.innerHTML = `<span>Longitude: ${mapLongitude.textContent}</span>`
+    windSpeed = measures === 'imperial' ? 'mph' : 'm/s'
+    searchInput.placeholder = 'Search city'
+    searchButton.innerText = 'search'
+    todayDescription.innerHTML = `<span>${description}</span>
+          <span>feels like: ${Math.round(feels_like)}°</span>
+          <span>wind: ${wind} ${windSpeed}</span>
+          <span>humidity: ${humidity}%</span>`
+      break
+  }
+}
```

Comment:
1) split the logic, each case is overloaded by it
2) create a constant with translations
3) set the default case for switch block